src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/JsonElementTests.cs

```diff
@@ -219,22 +220,35 @@ public static void DeepEquals_NotEqualValuesReturnFalse(string value1, string va
         }
 
         [Theory]
-        [InlineData(10)]
-        [InlineData(100)]
-        [InlineData(500)]
+        [InlineData(5)]
+        [InlineData(50)]
         public static void DeepEquals_DeepJsonDocument(int depth)
         {
             using JsonDocument jDoc = CreateDeepJsonDocument(depth);
             JsonElement element = jDoc.RootElement;
             Assert.True(JsonElement.DeepEquals(element, element));
         }
 
-        [Fact]
-        public static void DeepEquals_TooDeepJsonDocument_ThrowsInsufficientExecutionStackException()
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]
+        public static async Task DeepEquals_TooDeepJsonDocument_ThrowsInsufficientExecutionStackException()
         {
-            using JsonDocument jDoc = CreateDeepJsonDocument(10_000);
-            JsonElement element = jDoc.RootElement;
-            Assert.Throws<InsufficientExecutionStackException>(() => JsonElement.DeepEquals(element, element));
+            var tcs = new TaskCompletionSource<bool>();
+            new Thread(() =>
+            {
+                try
+                {
+                    using JsonDocument jDoc = CreateDeepJsonDocument(10_000);
+                    JsonElement element = jDoc.RootElement;
+                    Assert.Throws<InsufficientExecutionStackException>(() => JsonElement.DeepEquals(element, element));
+                    tcs.SetResult(true);
+                }
+                catch (Exception e)
+                {
+                    tcs.SetException(e);
+                }
+            }, maxStackSize: 100_000) { IsBackground = true }.Start();
```

Comment:
Is the mismatch in sizes intentional here?
```suggestion
            }, maxStackSize: 10_000) { IsBackground = true }.Start();
```

Reply:
I just picked something reasonably small. The most important aspect from my perspective is just having it be something specified, as otherwise the default is used, and the default differs across platforms, e.g. Linux is often an 8MB stack.